doc/kernel/usermode/mpu_stack_objects.rst

```diff
@@ -0,0 +1,49 @@
+.. _mpu_stack_objects:
+
+MPU Stack Objects
+#################
+
+Thread Stack Creation
+=====================
+
+Thread stacks are declared staticly with :c:macro:`K_THREAD_STACK_DEFINE()`
+or embedded within structures using c:macro:`K_THREAD_STACK_MEMBER()`
+
+For architectures which utilize memory protection unit (MPU) hardware,
+stacks are physically contiguous allocations.  This contiguous allocation
+has implications for the placement of stacks in memory, as well as the
+implementation of other features such as stack protection and userspace.
+
+Stack Guards
+============
+
+Stack protection mechanisms require hardware support that can restrict access
+to memory.  Memory protection units can provide this kind of support.
+The MPU provides a fixed number of regions.  Each region contains information
+about the start, end, size, and access attributes to be enforced on that
+particuliar region.
+
+Stack guards are implemented by using a single MPU region and setting the
+attributes for that region to not allow write access.  If invalid accesses
+occur, a fault ensues.  The stack guard is defined at the bottom of the stack,
+or the lowest address.
```

Comment:
To be clear, I think you're saying,

    The stack guard is defined at the bottom (the lowest address) of the stack.

You're use of "or" makes it sound like an alternative rather than clarifying what the "bottom of the stack" means.

Reply:
right.  i'll change it to your suggestion